src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -131,16 +134,43 @@ protected void Initialize(IDiscoveryContext context, IMessageLogger messageLogge
             {
                 Settings.Load(context);
                 TestLog.Verbosity = Settings.Verbosity;
-               
+                InitializeForbiddenFolders();
+                SetCurrentWorkingDirectory();
             }
             catch (Exception e)
             {
                 TestLog.Warning("Error initializing RunSettings. Default settings will be used");
                 TestLog.Warning(e.ToString());
             }
+            
+        }
+
+        public void InitializeForbiddenFolders()
+        {
+            ForbiddenFolders = new List<string>
+            {
+                Environment.GetEnvironmentVariable("ProgramW6432"),
+                Environment.GetEnvironmentVariable("ProgramFiles(x86)"),
+                Environment.GetEnvironmentVariable("windir"),
+            }.Where(o => !string.IsNullOrEmpty(o)).Select(o=>o.ToLower()+@"\").ToList();
+        }
+
+        private void SetCurrentWorkingDirectory()
+        {
+            var dir = Directory.GetCurrentDirectory();
+            bool foundForbiddenFolder = CheckDirectory(dir);
+            if (foundForbiddenFolder)
+                Directory.SetCurrentDirectory(Path.GetTempPath());
         }
 
-        
+        /// <summary>
+        /// If a directory matches one of the forbidden folders, then we should reroute, so we return true in that case
+        /// </summary>
+        public bool CheckDirectory(string dir)
+        {
+            var checkdir = (dir.EndsWith("\\") ? dir : dir + "\\").ToLower();
+            return ForbiddenFolders.Any(o => checkdir.StartsWith(o));
```

Comment:
I think there's a correctness issue with ToLower. The thing to do as I understand it is to never use ToLower or ToUpper (or their Invariant versions) but instead to use `checkdir.StartsWith(o, StringComparison.OrdinalIgnoreCase)`.

Reply:
http://www.moserware.com/2008/02/does-your-code-pass-turkey-test.html has more info

Reply:
Cool, that was a new one :-).  But done !